api/orchestration.go

```diff
@@ -2,339 +2,192 @@ package api
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
-	"regexp"
 
 	"github.com/YaleSpinup/apierror"
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/docdb"
 	log "github.com/sirupsen/logrus"
 )
 
-var DefaultAvailabilityZones = []*string{
-	aws.String("us-east-1a"),
-	aws.String("us-east-1d"),
-}
-
-var SubnetIDSlice = []*string{
-	aws.String("subnet-0707d40ddbb9d0818"),
-	aws.String("subnet-02364e7e9fc4d8045"),
-}
-
-/*
-// subnetAzs returns a map of subnets to availability zone names used by EFS
-// this may change to support getting the list of subnets as well, currently it uses
-// the defaults from the EFS service
-func (o *docDBOrchestrator) subnetAzs(ctx context.Context, account string) (map[string]string, error) {
-	log.Infof("determining availability zone for account %s using efs onezone", account)
+// documentDBCreate creates documentDB cluster and instances
+func (o *docDBOrchestrator) documentDBCreate(ctx context.Context, req *DocDBCreateRequest) (*DocDBResponse, error) {
+	log.Infof("creating documentDB cluster %s with %d instance(s)", aws.StringValue(req.DBClusterIdentifier), aws.IntValue(req.InstanceCount))
 
-	// 		DBSubnetGroupCreateStatus, err := o.client.CreateDBSubnetGroup(ctx, &inputDBSubnetGroup)
+	req.Tags = normalizeTags(o.org, req.Tags)
 
-	if !ok {
-		return nil, apierror.New(apierror.ErrNotFound, "account doesnt exist", nil)
+	// check if a DBSubnetGroup exists, and create it if needed
+	dbSubnetGroupFound, err := o.dbSubnetGroupExists(ctx, dbSubnetGroupName(o.org))
+	if err != nil {
+		return nil, err
+	} else if !dbSubnetGroupFound {
```

Comment:
We need the `else if` to check if the subnet group exists.
I wanted to explicitly log it as Info since it's useful to see which subnet group we are using exactly, especially in the beginning as we start using this

Reply:
> We need the else if to check if the subnet group exists.

unless I'm being dense (possible),  i don't think you do:

```golang
	// check if a DBSubnetGroup exists, and create it if needed
	dbSubnetGroupFound, err := o.dbSubnetGroupExists(ctx, dbSubnetGroupName(o.org))
	if err != nil {
		return nil, err
	}

        if !dbSubnetGroupFound {
		if err := o.dbSubnetGroupCreate(ctx, dbSubnetGroupName(o.org), req.SubnetIds); err != nil {
			return nil, err
		}
	} else {
		log.Infof("subnet group %s already exists, will use it for this docdb cluster", dbSubnetGroupName(o.org))
	}
```

> I wanted to explicitly log it as Info since it's useful to see which subnet group we are using exactly, especially in the beginning as we start using this

I thought you might be able to put the log message in `dbSubnetGroupExists()` instead since it would avoid the else, but this is ok too 👍

Reply:
maybe you were thinking to do this?

```golang
	// check if a DBSubnetGroup exists, and create it if needed
	if dbSubnetGroupFound, err := o.dbSubnetGroupExists(ctx, dbSubnetGroupName(o.org)); err != nil {
		return nil, err
	} else if !dbSubnetGroupFound {
		if err := o.dbSubnetGroupCreate(ctx, dbSubnetGroupName(o.org), req.SubnetIds); err != nil {
			return nil, err
		}
	} else {
		log.Infof("subnet group %s already exists, will use it for this docdb cluster", dbSubnetGroupName(o.org))
	}
```